Remove "Flask" label on Keyring API docs

alexandratran opened the pull request:
This PR removes the **Flask** label from the Keyring API guides and reference. Fixes #1154.

Preview: https://docs.metamask.io/1154-remove-flask-keyring/snaps/

AlexJupiter reviewed (commented):
@alexandratran thanks for making this. However I'm not convinced about this proposed change in page titles and navigation. Is is possible to provide more context on these changes beyond the removal of the Flask label or perhaps we split out into two separate PRs?

alexandratran commented:
@AlexJupiter This PR doesn't make any navigation/title changes. We updated the Snaps docs structure in https://github.com/MetaMask/metamask-docs/pull/1129, as requested by the Snaps team.

In that PR, we added a "Features" top-level category, adding "Custom EVM accounts" in that category. We changed the title "Use the Keyring API" to "Custom EVM accounts" because we want to highlight the features/use cases in that category, not just the tool/method that enables the feature.

That PR also consolidated all the accounts guides into one section (concept and how-tos) so readers don't have to jump around the docs to find all the relevant content.

alexandratran commented:
@AlexJupiter Also, I just noticed that we missed adding a redirect from [the old concepts page](https://docs.metamask.io/snaps/concepts/keyring-api/) to [the new page](https://docs.metamask.io/snaps/features/custom-evm-accounts/). Sorry if that confused you! I added the redirect in this PR.

AlexJupiter reviewed (approved):
Thanks @alexandratran this looks good now. That missing redirect did confuse me on my last review.
